### ISSUE TITLE
Removing and re-adding database with same folder path does not work properly

### ISSUE THREAD

**Issue by Marcono1234** [github/vscode-codeql#836]
**Version**
Extension 1.4.5
CodeQL CLI 2.5.1

**Describe the bug**
Removing a database after having run queries on it and viewed ASTs and then re-adding a different database at the same file system location (e.g. because you made changes to the source code and then created the database again) prevents browsing the source code of that database.

**To Reproduce**
1. Create a folder `test-db`
2. In that folder create a `Test.java` file with this content:
    ```java
    class Test {
    }
    ```
3. Run
    ```
    codeql database create --language=java --source-root=. "--command=javac *.java" db
    ```
4. In VSCode add the `test-db/db` folder as database
5. Run any query, e.g.:
    ```ql
    import java

    from Class c
    where c.fromSource()
    select c
    ```
6. Click on the result to open the `.java` source
7. Run "View AST"
8. Optional: Close the `.java` file and remove the AST view and the query result from the list
9. Use "Remove Database"
10. Make a change to the `Test.java` file:
    ```java
    class Test {
        void someMethod() { }
    }
    ```
11. Delete the `test-db/db` folder and create the database again:
    ```
    codeql database create --language=java --source-root=. "--command=javac *.java" db
    ```
12. In VSCode add the database again
13. Run any query again, and click on the result
:x: Nothing happens; the CodeQL Query Server output contains:
    ```
    Unable to handleMsgFromView: cannot open codeql-zip-archive://1-51/#REDACTED#%5Ctest-db%5Cdb%5Csrc.zip/#REDACTED#/test-db/Test.java. Detail: Unable to read file 'codeql-zip-archive://1-51/#REDACTED#\test-db\db\src.zip/#REDACTED#/test-db/Test.java' (Error: unexpected end of file)
    ```
    Similarly, clicking on the `Test.java` source file in the files explorer of VSCode causes a similar error.

Note: Steps 5 - 8 (inclusive) might not be needed to reproduce this.

**Expected behavior**
When re-adding a database for which a database with the same folder path existed previously, all state of that previous database should be removed.

Removal of previous database state could happen either when the database is removed, or when a database with the same folder path is re-added. Maybe it would be better to only perform this when re-adding the database (instead of on removal) to allow the user to still view some information after the database has been removed (might be useful when importing test databases).

**Comment by aeisenberg** [comment]
Does this work if you restart vscode after adding the database?

**Comment by Marcono1234** [comment]
Yes restarting VSCode can be used as workaround, though it is rather cumbersome.

**Comment by aeisenberg** [comment]
Yes, that's not a reasonable solution. Just wanted to know if the state was persistent.
